libraries/AP_ICEngine/AP_ICEngine.cpp

```diff
@@ -167,6 +167,12 @@ const AP_Param::GroupInfo AP_ICEngine::var_info[] = {
 
     // 18 was IGNITION_RLY
 
+    // @Param: MAX_RETRY
+    // @DisplayName: Maximum number of retries
+    // @Description: If set 0 then there is no limit to retrials. If set to a value greater than 0 then the engine will retry starting the engine this many times before giving up.
+    // @User: Standard
+    // @Range: 0 254
```

Comment:
Underlying type as Int8, not UInt8. Should change this to a max of 127 or should change to uint8

Reply:
will fix it.